README.md

```diff
@@ -1,41 +1,149 @@
+      build:
+        config:
+          # All config defined here is passed to the __init__ function of the corresponding service adapter.
+          steps:
+            - type: build
+              args:
+                image: demo-image
+                dockerfile: Dockerfile
+      deploy:
+        commands:
+          - docker login -u _json_key -p "$(cat /path/to/keyfile.json)" https://us.gcr.io/my-gcp-project
+        config:
+          # All config defined here is passed to the __init__ function of the corresponding service adapter.
+          registry_path: my-gcp-project/this-project
+          steps:
+            - type: push
+              args:
+                image: demo-image
+                tags:
+                  - latest
+...
+```
+There are two types of providers: those that build artifacts from code in the repository Servicer is working within (these are in `servicer/builtin/service_adapters/package`), and those that focus on interacting with external systems, like Google Kubernetes Clusters (other subdirectories of `servicer/builtin/service_adapters`).
+
+In general, there are two types of service adapters (not including `provider`s): service adapters that connect Servicer with third parties, like Google Cloud Platform, or Amazon Web Services; and the `package` provider, which includes tools for building artifacts. Below is a service that builds a python package, and deploys it to Artifactory.
+
+```
+  my_python_example:
+    provider: package
+    service_type: pypi
+    depends_on:
+      - artifactory-pypi
+      - pypi-credentials
+    steps:
+      build:
+        commands:    # These are standard shell commands Servicer will execute
+          - rm -rf python_example/dist
+          - python setup.py sdist
+        config:
+          # The service adapter handles things nested at this level and deeper
+          package_info:
+            package_file_path: setup.py
+          steps:
+            - type: read_package_info
+      deploy:
+          config:
+            package_info:
+              package_file_path: setup.py
+            steps:
+              - type: upload
+                args:
+                  server: artifactory
+                  path: dist/*.tar.gz
+```
+
+Using `depends_on` defines the dependencies for a service. During each step, Servicer will execute it on the dependees before executing it on the dependors. In this case, during the `build` step, Servicer will execute it on `artifactory-pypi` and `pypi-credentials` before executing the `build` step for `my_python_package`.
+
+Dependors can be either services that are defined elsewhere in `services.yaml`, or service adapters you've defined in `.servicer/service_adapters` in your repository. In this example, `artifactory-pypi` and `pypi-credentials` are service adapters that do not come with Servicer, and were added by the user.
+
+
+```
```

Comment:
`artifactory-pypi` and `pypi-credentials` do indeed come builtin to Servicer, they are the `artifactory/artifactory.py` and `package/pypi.py` Service Adapters.

Fundamentally, `depends_on` will only accept service names, so dependees can only be the services defined in `services.yaml`

Not sure what "it" is referring to  in "Servicer will execute it", what about this?

Using `depends_on` defines the other services your service depends on. Servicer will execute the dependee's matching step before executing the dependor's current step. In this case, Servicer will first execute the `build` steps for `artifactory-pypi` and `pypi-credentials` before executing the `build` step for `my_python_package`.